File: src/test/java/de/spries/fleetcommander/model/facade/OwnPlayerTest.java
package de.spries.fleetcommander.model.facade;

import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.verify;

import org.junit.Before;
import org.junit.Test;

import de.spries.fleetcommander.model.core.Player;

public class OwnPlayerTest {

	private Player originalPlayer;
	private OwnPlayer viewingPlayer;

	@Before
	public void setUp() {
		originalPlayer = mock(Player.class);
		viewingPlayer = new OwnPlayer(originalPlayer);
	}

	@Test
	public void forwardsCallToGetNameForOtherPlayers() {
		viewingPlayer.getCredits();
		verify(originalPlayer).getCredits();
	}
}

File: src/test/kotlin/de/spries/fleetcommander/model/facade/OwnPlayerTest.kt
package de.spries.fleetcommander.model.facade

import com.nhaarman.mockitokotlin2.mock
import de.spries.fleetcommander.model.core.Player
import org.junit.Before
import org.junit.Test
import org.mockito.Mockito.verify

class OwnPlayerTest {

    private lateinit var originalPlayer: Player
    private lateinit var viewingPlayer: OwnPlayer

    @Before
    fun setUp() {
        originalPlayer = mock()
        viewingPlayer = OwnPlayer(originalPlayer)
    }

    @Test
    fun forwardsCallToGetNameForOtherPlayers() {
        viewingPlayer.getCredits()
        verify(originalPlayer).getCredits()
    }

    @Test
    @Throws(Exception::class)
    fun forwardsCallToCanAffordFactory() {
        viewingPlayer.getCanAffordFactory()
        verify(originalPlayer).canAffordFactory()
    }
}
